### PR TITLE
Allow bypassing authentication

### LINKED ISSUE #3459: Support capped collection clean-up
### What should be done?

After #3466, #3467, and #3498, we should add support for cleaning up capped collections.

Let's start a background goroutine in the handler that would do the following every minute (configurable with a test flag) if `EnableOplog` is `true`:
* iterate over a list of databases;
* iterate over a list of capped collections in those databases;
* run `Collection.Stats` on the current collection;
* if stats are lower than capped collection parameters, continue to the next collection;
* query 10% (configurable with a test flag) RecordIDs of documents in this collection (with `Limit` and `OnlyRecordIDs` set);
* delete them;
* compact that collection.

The last step essentially involves the calling of the `compact` command with `force: false` (#3466). We should not call that command handler directly but refactor it to make the shared code callable from the handler and our background goroutine.

Databases and collections could disappear during the iteration; that should be handled.

The goroutine should not `panic` and not `recover`.
We should add basic logging and metrics for that background goroutine.

Most of the logic should already be tested. No new automated tests are necessary.

### Where?

The handler. All backend changes should already be done by that point.

### Definition of Done

- handlers updated;
- all backends tested manually;
- spot refactorings done;
- user documentation checked and updated.

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #3459.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — APPROVED** [review]
LGTM.

Side question: so, we always store every new username-password group in the connection pool. But we don't cleanup the pool, right? E.g. if 100 users connect with different usernames and passwords, we will store them all unless FerretDB is stopped?

File: internal/backends/postgresql/metadata/pool/pool.go [hunk 1]
```diff
@@ -134,6 +134,22 @@ func (p *Pool) Get(username, password string) (*pgxpool.Pool, error) {
 	return res, nil
 }
 
+// GetAny returns a random open pool of connections to PostgreSQL, or nil if none are available.
+func (p *Pool) GetAny() *pgxpool.Pool {
+	p.rw.RLock()
+	defer p.rw.RUnlock()
+
+	for _, pool := range p.pools {
```
Comment (henvic) [inline internal/backends/postgresql/metadata/pool/pool.go:142]: I think this clever approach of getting a first map element might backfire as the same element might be returned multiple times (as in, with a bad distribution). What do you think?

Comment (AlekSi) [reply internal/backends/postgresql/metadata/pool/pool.go:142]: Well, more reasons to finish authentication faster 😆 

Seriously, though, the runtime actually makes effort to randomize iterations over maps, so it is as bad as map implementation (which is good enough)

**Comment by AlekSi** [comment]
> we always store every new username-password group in the connection pool.

Pools.

> But we don't cleanup the pool, right?

Right, we don't clean-up pools.

> E.g. if 100 users connect with different usernames and passwords, we will store them all unless FerretDB is stopped?

If there are 100 _valid_ username/password combinations, then yes,  in that case we store 100 strings and 100 pointer. Each pointer point to a pool of connections which automatically scales down to zero: https://github.com/FerretDB/FerretDB/blob/6737d3260e4059182e8fe59af64a3549dcb2ecf9/internal/backends/postgresql/metadata/pool/uri.go#L28-L31

**Review by henvic — APPROVED** [review]

**Comment by xaitx** [comment]
Why would this cause authentication bypass

**Comment by AlekSi** [comment]
Authentication is bypassed when `ConnInfo`'s (object representing client connection information) field `BypassAuth` is set to true. It is not set for normal connections, but needed for background workers such as capped collection clean-up (#3459).

**Pull request merged**